addons/event/models/event_registration.py

```diff
@@ -179,11 +169,18 @@ def name_get(self):
             ret_list.append((registration.id, name))
         return ret_list
 
+    def toggle_active(self):
+        pre_inactive = self - self.filtered(self._active_name)
+        super().toggle_active()
+        # Necessary triggers as changing registration states cannot be used as triggers for the
+        # Event(Ticket) models constraints.
+        if pre_inactive:
+            pre_inactive.mapped('event_id')._check_seats_availability()
+            pre_inactive.mapped('event_ticket_id')._check_seats_availability()
+
     def _check_auto_confirmation(self):
-        if any(not registration.event_id.auto_confirm or
-               (not registration.event_id.seats_available and registration.event_id.seats_limited) for registration in self):
-            return False
-        return True
+        """ Checks that all registrations are for `auto-confirm` events. """
+        return self.mapped('event_id.auto_confirm') == [True]
```

Comment:
Isn't "mapped" adding a value for each registration ? If we create registrations in batch on several events (import mode) what happens ? Shouldn't we do something like
```
return all(event.auto_confirm for event in self.event_id)
```